pkg/stanza/fileconsumer/file.go

```diff
@@ -109,162 +130,161 @@ func (m *Manager) startPoller(ctx context.Context) {
 func (m *Manager) poll(ctx context.Context) {
 	// Increment the generation on all known readers
 	// This is done here because the next generation is about to start
+	m.knownFilesLock.Lock()
 	for i := 0; i < len(m.knownFiles); i++ {
 		m.knownFiles[i].generation++
 	}
-
-	// Used to keep track of the number of batches processed in this poll cycle
-	batchesProcessed := 0
+	m.knownFilesLock.Unlock()
 
 	// Get the list of paths on disk
 	matches := m.finder.FindFiles()
-	for len(matches) > m.maxBatchFiles {
-		m.consume(ctx, matches[:m.maxBatchFiles])
+	m.consume(ctx, matches)
+	m.clearCurrentFiles()
 
-		// If a maxBatches is set, check if we have hit the limit
-		if m.maxBatches != 0 {
-			batchesProcessed++
-			if batchesProcessed >= m.maxBatches {
-				return
+	// Any new files that appear should be consumed entirely
+	m.readerFactory.fromBeginning = true
+	m.syncLastPollFiles(ctx)
+}
+
+func (m *Manager) clearCurrentFiles() {
+	m.currentFiles = make([]*os.File, 0)
+	m.currentFps = make([]*Fingerprint, 0)
+}
+
+func (m *Manager) worker(ctx context.Context) {
+	defer m.workerWg.Done()
+	for {
+		chanData, ok := <-m.readerChan
+
+		if !ok {
+			return
+		}
+		r, path := chanData.reader, chanData.path
+		r.ReadToEnd(ctx)
+		// Delete a file if deleteAfterRead is enabled and we reached the end of the file
+		if m.deleteAfterRead && r.eof {
+			r.Close()
+			if err := os.Remove(r.file.Name()); err != nil {
+				m.Errorf("could not delete %s", r.file.Name())
 			}
+		} else {
+			// Save off any files that were not fully read or if deleteAfterRead is false
+			m.saveCurrent([]*Reader{r})
+			m.readerLock.Lock()
+			m.lostReaders = append(m.lostReaders, r)
```

Comment:
This slice stores the readers once they are consumed.
In subsequent poll cycles, these will be used to keep track of files which are lost.
This will be used in `handleLostFiles` function before every poll cycle starts.